app/services/claims/import_schools.rb

```diff
@@ -8,11 +8,32 @@ def initialize(csv_file_path:)
   end
 
   def call
-    urns = CSV.read(csv_file_path, headers: true).map { |row| row["urn"] }
+    urns = []
+    users_data = []
 
-    updated_count = School.where(urn: urns).update_all(claims_service: true)
+    CSV.foreach(csv_file_path, headers: true) do |row|
+      urns << row["placement_school_urn"]
 
-    Rails.logger.info "#{updated_count} schools updated."
+      users_data << {
+        first_name: row["First name"],
+        last_name: row["Last name"],
+        email: row["Email"],
+        school_urn: row["placement_school_urn"],
+      }
+    end
+
+    schools = School.where(urn: urns)
+    schools.update_all(claims_service: true)
+
+    users_data.each do |data|
+      school = schools.detect { |s| s.urn == data[:school_urn] }
+
+      begin
+        Claims::User.create!(first_name: data[:first_name], last_name: data[:last_name], email: data[:email], user_memberships: [UserMembership.new(organisation: school)])
+      rescue ActiveRecord::RecordInvalid => e
+        Rails.logger.error "Failed to create user: #{e.message}"
+      end
+    end
```

Comment:
It's not letting me put in a suggestion, but how about:

```ruby
csv_string = # we'll be decrypting and reading from memory, so no need to base arguments on a path. If a path, it will be the encrypted file path and we'll need to decrypt here.

# Ensure all are added or none at all
ApplicationRecord.transaction do
  CSV.parse(csv_string, header: true) do |row|
    school = School.find_by!(urn: row["placement_school_urn"])

    school.update!(claims_service: true)
    school.users << Claims::User.create!(
      first_name: row["First name"],
      last_name: row["Last name"],
      email: row["Email"],
    )
  end
end
```

This updates each School slower, but this service should be infrequently run. For our dataset, it should be fine.